src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.Streaming.cs

```diff
@@ -311,6 +319,75 @@ private void WriteComponentHtml(int componentId, TextWriter output, bool allowBo
         }
     }
 
+    private void AppendWebAssemblyPreloadHeaders()
+    {
+        var assets = _httpContext.GetEndpoint()?.Metadata.GetMetadata<ResourceAssetCollection>();
+        if (assets != null)
+        {
+            var headers = new List<(string? Order, string Value)>();
+            foreach (var asset in assets)
+            {
+                if (asset.Properties == null)
+                {
+                    continue;
+                }
+
+                // Use preloadgroup=webassembly to identify assets that should to be preloaded
+                string? header = null;
+                foreach (var property in asset.Properties)
+                {
+                    if (property.Name.Equals("preloadgroup", StringComparison.OrdinalIgnoreCase) && property.Value.Equals("webassembly", StringComparison.OrdinalIgnoreCase))
+                    {
+                        header = $"<{asset.Url}>";
+                        break;
+                    }
+                }
+
+                if (header == null)
+                {
+                    continue;
+                }
+
+                string? order = null;
+                foreach (var property in asset.Properties)
+                {
+                    if (property.Name.Equals("preloadrel", StringComparison.OrdinalIgnoreCase))
+                    {
+                        header = String.Concat(header, "; rel=", property.Value);
```

Comment:
Use StringBuilder

Reply:
I ended up with StringBuilder, because not all of the attributes are actually required, so it would result in a nested interpolated string. I could at let compute the StringBuilder length, but since we are talking about 6 items at the moment, I'm not sure if it's worth the effort.

Let me know if there's a better approach

Reply:
> Even better I think, an interpolated string, as that avoids any allocation at all.

I think they are not better than `StringBuilder`. To make them comparable you would have to use `InterpolatedStringHandler`, right ?

If we are caching this, that level of optimization is not worth it.